src/instructlab/sdg/pipeline.py

```diff
@@ -131,15 +131,21 @@ def from_file(cls, ctx, pipeline_yaml):
             pipeline_yaml = os.path.join(resources.files(__package__), pipeline_yaml)
         return cls(ctx, pipeline_yaml, *_parse_pipeline_config_file(pipeline_yaml))
 
-    def generate(self, dataset) -> Dataset:
+    def generate(self, dataset, leaf_node_path=None) -> Dataset:
```

Comment:
I think it would make sense to pass a `checkpoint_dir` arg to a generate method. This makes it more generic and aligned to other ML libraries. 

Also in general would be good to not use `leaf_node_path` or any other var names which are tied to the way we have our generate_data setup.

Reply:
> I think it would make sense to pass a `checkpoint_dir` arg to a generate method. This makes it more generic and aligned to other ML libraries.

checkpoint_dir is already being passed via the ctx.  So I figured it would make since to only pass the additional info.  If we are going to pass the entire checkpoint_dir, we'll probably want to remove the root from the ctx.

> Also in general would be good to not use leaf_node_path or any other var names which are tied to the way we have our generate_data setup.

Will rename

Reply:
`checkpoint_dir` is not used elsewhere other than `.generate` method. So does it make sense to have it tied up with the `PipelineContext`? 

I would suggest something like this: 

```python
def generate(self, dataset, checkpoint_dir=None) -> Dataset:
```

where `checkpoint_dir` is the full path (which includes the specific leaf_node)

Reply:
PipelineContext contains runtime parameters that do not differ over calls to 'generate()' - the base directory for checkpoints fits within that category. A name to identify a particular pipeline run (i.e. a call to generate()) makes sense as a parameter to 'generate()'